code/modules/mob/living/living_defense.dm

```diff
@@ -68,6 +68,23 @@
 		else
 				return 0
 
+/mob/living/proc/set_combat_mode(new_mode, silent)
+	if(combat_mode == new_mode)
+		return
+	combat_mode = new_mode
+	client?.show_popup_menus = !src.combat_mode
+	if(hud_used?.action_intent)
+		hud_used.action_intent.update_icon()
+	if(silent)
+		return
+	if(combat_mode)
+		playsound_local(src, 'sound/misc/ui_togglecombat.ogg', 50, FALSE, pressure_affected = FALSE) //Sound from interbay!
+	else
+		playsound_local(src, 'sound/misc/ui_toggleoffcombat.ogg', 50, FALSE, pressure_affected = FALSE) //Slightly modified version of the above
+
+///Getter for possibly extending behavior in the future
+/mob/living/proc/in_combat_mode()
+	return combat_mode
```

Comment:
```suggestion
#define IN_COMBAT_MODE(target) target.combat_mode
```
Avoid extending behavior on `is_thing()` checks. We don't want this kind of check redefined ever. If something is actually not in combat mode, then the associated variable should represent it, so we can track when it enters and leaves combat mode. Else we can't react to the event.

Reply:
sure, I'll do a big old regex to fix it in a bit.